### PR TITLE
reverse-string: rename func String to Reverse

### PR DESCRIPTION
**Pull request by sbromberger**
Renames `String()` to `Reverse()`.

### CODE REVIEW

**Comment by bitfield** [comment]
LGTM! If there's no objection from anyone else I'll merge this shortly.

File: exercises/reverse-string/example.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package reverse
```
Comment (leenipper) [inline exercises/reverse-string/example.go]: I remember this one.  See https://github.com/exercism/go/pull/971
and the discussion about the func name.
In actual client code (of which there is none for an exercise), you would have reverse.String(x). With the change, a caller would use reverse.Reverse(x). A bit wordy, but I understand the reason for the PR.
@bitfield any new thoughts on this one ?

**Comment by da-edra** [comment]
I think the commit message should be amended to be more descriptive because if you're browsing through the history of the repos a commit saying `implements #1230` doesn't say much.

**Comment by sbromberger** [comment]
Hi,

I'm no longer in a position to support this PR. If someone wants to pick it up, great; otherwise, feel free to close it.

**Comment by bitfield** [comment]
I see @leenipper's point about the `reverse.Reverse` stutter, but I also think this makes a persuasive case:

```go
-		return s == String(String(s))
+		return s == Reverse(Reverse(s))
```

With an improved commit message, I'm 👍.

**Comment by leenipper** [comment]
Ok @bitfield, I'm convinced. I'll change the PR title to:

reverse-string: rename func String to Reverse

Better?

It's a single commit, so I think a Squash and Merge will let the PR title be the commit title. Or is there a better way?  I'm expecting that a Squash and Merge won't alter the original commit author info.

And I would add a "Resolves #1230"

**Comment by leenipper** [comment]
I was mistaken; it looks like a Squash and Merge changes the commit author. I get that impression from the github UI.  I'm not sure there's a way to update the commit subject line while keeping @sbromberger as author, other than a local commit --amend and force push by @sbromberger.  Enlighten me if there is a way.

**Comment by sbromberger** [comment]
I don't care about being an author. It's a simple enough change that someone else can probably just make it. I'd recommend just closing this out and making a new PR.
